drivers/sx1280/sx1280.c

```diff
@@ -0,0 +1,413 @@
+        DEBUG("[sx1280] init: ERROR failed irq disabled, %d\n", res);
+        return -EIO;
+    }
+
+    sx1280_set_rx(dev->ral.context, SX1280_TICK_SIZE_1000_US, SX1280_RX_CONTINUOUS_MODE );
+    int16_t rssi = 0;
+
+    for (unsigned i = 0; i < 32; i++) {
+        ztimer_sleep(ZTIMER_MSEC, 1);   /* wait one millisecond */
+        sx1280_get_rssi_inst(dev->ral.context, &rssi );
+        /* Non-filtered RSSI value reading. Only takes the LSB value */
+        rnd |= ((uint32_t)rssi & 0x01) << i;
+    }
+
+    enable_irq(dev);
+    ral_sx1280_set_standby(&dev->ral);
+    return rnd;
+}
+
+void sx1280_setup(sx1280_t *dev, const sx1280_params_t *params, uint8_t index)
+{
+    dev->ral.context = dev;
+    dev->ral.radio_type = RAL_RADIO_SX1280;
+    dev->ral.tcxo_cfg.tcxo_ctrl_mode = RAL_TCXO_NONE;
+
+    netdev_t *netdev = &dev->netdev;
+
+    netdev->driver = &sx1280_driver;
+    dev->params = (sx1280_params_t *)params;
+    netdev_register(&dev->netdev, NETDEV_SX1280, index);
+}
+
+static uint32_t sx1280_init_default_config(sx1280_t *dev)
+
+{
+    /* default parameters */
+    const ral_params_lora_t params_default = {
+        .freq_in_hz = SX1280_CHANNEL_DEFAULT,
+        .sf = _lora_sf_to_ral_sf(CONFIG_LORA_SF_DEFAULT),
+        .bw = _lora_bw_to_ral_bw(CONFIG_LORA_BW_DEFAULT),
+        .cr = _lora_cr_to_ral_cr(CONFIG_LORA_CR_DEFAULT),
+        .pbl_len_in_symb = CONFIG_LORA_PREAMBLE_LENGTH_DEFAULT,
+        .sync_word = LORA_SYNCWORD_ISM2400_PUBLIC,
+        .pld_is_fix = false,
+        .crc_is_on = true,
+        .invert_iq_is_on = false,
+        .symb_nb_timeout = 0,
+        .pwr_in_dbm = SX1280_RADIO_TX_POWER,
+    };
+
+    /* set current ral_params */
+    dev->ral_params_lora = params_default;
+
+    ral_status_t res = ral_sx1280_setup_lora(&dev->ral, &params_default);
+
+    if (res != RAL_STATUS_OK) {
+        DEBUG("[sx1280] init: ERROR failed LoRa setup, %d\n", res);
+        return -EIO;
+    }
+
+    return 0;
+}
+
+int sx1280_init(sx1280_t *dev)
+{
+    /* Setup SPI for SX1280 */
+    int res = spi_init_cs(dev->params->spi, dev->params->nss_pin);
+
+    if (res != SPI_OK) {
+        DEBUG("[sx1280] error: failed to initialize SPI_%i device (code %i)\n",
+              dev->params->spi, res);
+        return -1;
+    }
+
+    DEBUG("[sx1280] init: SPI_%i initialized with success\n", dev->params->spi);
+    /* Initialize Reset */
+    if (gpio_init(dev->params->reset_pin, GPIO_OUT) < 0) {
+        DEBUG("[sx1280] error: failed to initialize RESET pin\n");
+        return -EIO;
+    }
+    /* Initialize DIO0s */
+    if (gpio_init(dev->params->dio0_pin, GPIO_IN) < 0) {
+        DEBUG("[sx1280] error: failed to initialize DIO1 pin\n");
+        return -EIO;
+    }
+    /* Initialize DI01s */
+    if (gpio_init_int(dev->params->dio1_pin, GPIO_IN, GPIO_RISING, _dio1_isr, dev)) {
+        DEBUG("[sx1280] error: failed to initialize DIO1 pin\n");
+        return -EIO;
+    }
+    ztimer_sleep(ZTIMER_MSEC, 2);
```

Comment:
Maybe comment why we need to sleep for 2ms?  All I could find was
![image](https://user-images.githubusercontent.com/19396439/167562240-c91a0d67-07b4-44c7-929a-8bb82ac097a5.png)

Unless there are some timeouts I am unaware of?